pkg/util/chunkinfologger/debugheader.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkinfologger
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/dskit/middleware"
+)
+
+type chunkInfoLoggingContextType int
+
+const chunkInfoLoggingContextKey = chunkInfoLoggingContextType(1)
+
+const ChunkInfoLoggingHeader = "X-Mimir-Chunk-Info-Logger"
+
+func Middleware() middleware.Interface {
+	return middleware.Func(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+			if value := r.Header.Get(ChunkInfoLoggingHeader); value != "" {
+				// Split along commas
+				labels := strings.Split(value, ",")
+				ctx = ContextWithChunkInfoLogging(ctx, labels)
```

Comment:
Would it be worthwhile having a wildcard option as well, for logging all the series labels without having to list all of them?

Reply:
sure, in a next PR ;)